src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory_Methods.vb

```diff
@@ -439,6 +439,28 @@ Namespace Microsoft.CodeAnalysis.Semantics
                 [event], instance, Create(statement.Handler), adds:=False, isInvalid:=statement.HasErrors, syntax:=statement.Syntax, type:=Nothing, constantValue:=Nothing)
         End Function
 
+        Private Function GetConversionKind(kind As VisualBasic.ConversionKind) As Semantics.ConversionKind
+            Dim operationKind = Semantics.ConversionKind.Invalid
+
+            If kind.HasFlag(VisualBasic.ConversionKind.UserDefined) Then
+                operationKind = Semantics.ConversionKind.OperatorMethod
+            ElseIf Conversions.IsIdentityConversion(kind) Or
+                   kind.HasFlag(VisualBasic.ConversionKind.Reference) Or
+                   kind.HasFlag(VisualBasic.ConversionKind.TypeParameter) Or
+                   kind.HasFlag(VisualBasic.ConversionKind.Array) Or
+                   kind.HasFlag(VisualBasic.ConversionKind.Value) Then
+                operationKind = Semantics.ConversionKind.Cast
```

Comment:
TryCast has a larger bug in both languages, that I think will need to be resolved by removing it from the ConversionKind altogether and making it a field on IConversion. For example, we currently treat this as a `TryCast`:
`var i = 1.0 as int?;`
However, that's an explicit nullable conversion, which are saying a conversion with C# semantics when it's performed like this:
`var i = (int?)1.0;`
I don't think this is resolvable, and we should have a field on IConversion that specifies whether the cast will throw an exception if it fails or just return null.